src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -54,52 +51,106 @@ public ContigAligner(final String referenceFilePath) throws IOException {
-                // todo: parse cigar for internal indels
-                if (alnRgns.length > 1) {
-
-                    // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
-                    final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
-                            .filter(a -> a.getSecondary() < 0)
-                            .map(AlignmentRegion::new)
-                            .sorted(Comparator.comparing(a -> a.startInAssembledContig))
-                            .collect(arrayListCollector(alnRgns.length));
-
-
-                    final Iterator<AlignmentRegion> iterator = alignmentRegionList.iterator();
-                    if ( iterator.hasNext() ) {
-                        AlignmentRegion current = iterator.next();
-                        while ( iterator.hasNext() ) {
-                            final AlignmentRegion previous = current;
-                            current = iterator.next();
-                            String homology = "NA";
-                            if (previous.endInAssembledContig >= current.startInAssembledContig) {
-                                homology = new String(Arrays.copyOfRange(sequence, current.startInAssembledContig - 1, previous.endInAssembledContig));
-                            }
-
-                            String insertedSequence = "NA";
-                            if (previous.endInAssembledContig < current.startInAssembledContig - 1) {
-                                insertedSequence = new String(Arrays.copyOfRange(sequence, previous.endInAssembledContig, current.startInAssembledContig));
-                            }
-                            final AssembledBreakpoint assembledBreakpoint = new AssembledBreakpoint(contigId, previous, current, homology, insertedSequence);
-                            assembledBreakpoints.add(assembledBreakpoint);
-                        }
-                    }
-                }
+                // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
+                final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
+                        .filter(a -> a.getSecondary() < 0)
+                        .map(a -> new AlignmentRegion(breakpointId, contigId, a))
+                        .sorted(Comparator.comparing(a -> a.startInAssembledContig))
+                        .collect(arrayListCollector(alnRgns.length));
+                 alignedContigs.addAll(alignmentRegionList);
             }
         } catch (final IOException e) {
             throw new GATKException("could not execute BWA");
         }
 
-        return assembledBreakpoints;
+        return alignedContigs;
+    }
+
+    @VisibleForTesting
+    public static List<AssembledBreakpoint> getAssembledBreakpointsFromAlignmentRegions(final byte[] sequence, final List<AlignmentRegion> alignmentRegionList, final Integer minAlignLength) {
```

Comment:
This method is only used in `CallVariantsFromAlignedContigsSpark`. Would it make sense to move it there?
And the next three utility methods seem to only serve this method, so they may be moved together with this, if you decide to move them.